### ISSUE TITLE
How to understand the centers shown in the LMCL_loss?

### ISSUE THREAD

**Issue by taylover-pei** [YirongMao/softmax_variants#4]
You have done a great work.

But I have a question about the LMCL_loss. How to understand the centers as show bellow:

![image](https://user-images.githubusercontent.com/27360803/65816673-69ee2380-e231-11e9-878d-b8f695001cf3.png)

I think thay are stand for the parameters of the last fully connected layer. But I wonder how to upodate them. In your code, I think they are fixed values, are they?

Looking forward to your reply.

**Comment by YirongMao** [comment]
In here, centers are nn.Parameter(). They will updated by SGD or SGD variant methods. 
More specifically, the gradient of centers will be computed via 'loss.backward' as shown in https://github.com/YirongMao/softmax_variants/blob/master/train_mnist_LMCL.py#L71

You can also print out net.parameters() to check  whether centers are optimized in line https://github.com/YirongMao/softmax_variants/blob/master/train_mnist_LMCL.py#L117
